packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -334,11 +336,67 @@ abstract class IntelliJValidator extends DoctorValidator {
       return;
     }
 
+    
+
     messages.add(new ValidationMessage.error(
       '$title plugin not installed; this adds $title specific functionality.'
     ));
   }
 
+  Future<Null> _validateFlutterPackage(List<ValidationMessage> messages, List<String> packageNames, String title, {
+    Version minVersion
+  }) async {
+
+    for (String packageName in packageNames) {
+      if (!hasPackage(packageName)) {
+        continue;
+      }
+      final String versionText = _readPackageVersion(packageName);
+      final Version localVersion = new Version.parse(versionText);
+      if (localVersion != null && minVersion != null && localVersion < minVersion) {
+        messages.add(new ValidationMessage.error(
+          '$title plugin version $versionText - the recommended minimum version is $minVersion'
+        ));
+      } 
+      else {
+        
+        try {
+          final String latestFlutter = await _getLatestVersionPossible();
+          final Version flutterVersion = new Version.parse(latestFlutter);
+          final String serverMessage = localVersion < flutterVersion ? "[Consider upgrading your plugin version to $version]" : "";
+            messages.add(new ValidationMessage(
+              '$title plugin ${localVersion != null ? "version $localVersion" : "installed"} $serverMessage'
+            ));
+        } catch (e) {
+          messages.add(new ValidationMessage(
+            '$title plugin ${localVersion != null ? "version $localVersion" : "installed"}'
+          ));
+        }
+      }
+
+      return;
+    }
+    messages.add(new ValidationMessage.error(
+      '$title plugin not installed; this adds $title specific functionality.'
+    ));
+  }
+
+  Future<String> _getLatestVersionPossible() async {
+      
+      final List<String> stripVersion = version.split(".");
```

Comment:
Would it be too defensive to add a check to ensure `stripVersion.length >= 3`?

@alexander-doroshko : is this a future-safe approach to parsing building number?

Reply:
I'm going to try to modify the intellij validator that optains the version from the directory of intellij and obtain the build number from there. Then it would be probably more future proof.

Reply:
I guess this code turns `2017.3.1` into `173.1`. At a quick glance, it will fail for `2017.3`, which was the first in the train, and for Android Studio, which uses different versioning (e.g. `3.0`).
If might fail for older versions of IntelliJ IDEA, prior to 20XX.X versioning scheme.
Also, guessing a build number drops the possibility of fine-tuning the plugin's since/until constraints.

True build number can be found in the `build.txt` file inside the IDE installation. I guess Android Studio also has it, need to check. Make sure not to fail if the file contains version like `181.SNAPSHOT`.

Reply:
`build.txt` is located right in the installation folder on Windows. I guess the same on Linux and probably inside Contents folder on Mac. 
Well, the testing matrix for this feature is pretty big.

Reply:
Ok i got it to obtain the Build number exact from the plist file on Mac that is how it gets its version number. let me see if it is as easy on WindowsAndLinux